src/os/FileStore.cc

```diff
@@ -4180,7 +4196,8 @@ int FileStore::_setattrs(coll_t cid, const ghobject_t& oid, map<string,bufferptr
 	continue;
     }
     omap_remove.insert(p->first);
-    inline_set.insert(*p);
+    if (!inline_set.count(p->first))
+      inline_set.insert(make_pair(p->first, make_pair(p->second, -1)));
```

Comment:
would it be simpler to switch back to multiple setattr ops (instead of a single setattrs op)?  I don't want to make this code too complex...